numpy/ma/core.pyi

```diff
@@ -1841,7 +1847,13 @@ class MaskedArray(ndarray[_ShapeT_co, _DTypeT_co]):
     def tofile(self, /, fid: Incomplete, sep: str = "", format: str = "%s") -> Incomplete: ...
 
     #
-    def __reduce__(self): ...
+    def __reduce__(self) -> tuple[
+        Callable[[PickleArgs], MaskedArray],
+        PickleArgs,
+        Incomplete  # output of (untyped) `self.__getstate__`
```

Comment:
if you wanna get fancy about it (which really isn't needed here), I guess you could use a 

```pyi
@type_check_only
class CanGetState[+StateT](Protocol):
    def __getsate__(self, /) -> StateT: ...
```

so that you can 

```pyi
    def __reduce__[StateT](self: CanGetState[StateT], /) -> tuple[
        Callable[[PickleArgs], MaskedArray],
        PickleArgs,
        StateT,
    ]
```

but that's probably a bit overengineer-y in this case :P